### PR TITLE
ci: run missing tests

### PR DESCRIPTION
**Pull request by lgfa29**
The packages added to `ci/test-core.json` were not being tested in this release branch. Futhermore, the `client/vaultclient` package had an import cycle that was only fixed in `main`.

### CODE REVIEW

File: ci/test-core.json [hunk 1]
```diff
@@ -26,11 +26,13 @@
     "client/stats/...",
     "client/structs/...",
     "client/taskenv/...",
+    "client/vaultclient/...",
     "command/agent/...",
     "command/ui/...",
     "helper/...",
     "internal/...",
     "jobspec/...",
+    "jobspec2/...",
```
Comment (tgross) [inline ci/test-core.json:35]: Looks like there are some test failures associated with this package in this branch.

Comment (lgfa29) [reply ci/test-core.json:35]: Ah yes, I was waiting for CI to finish and forgot to mark this as draft. I will get the tests fixed today.

Comment (lgfa29) [reply ci/test-core.json:35]: Tests fixed in https://github.com/hashicorp/nomad/pull/19959/commits/8c6396994d365050ebfb347340f1c04c2f1688d2 by backporting the relevant parts of https://github.com/hashicorp/nomad/pull/19024.

**Review by gulducat — APPROVED** [review]
:test_tube: :tada:

**Pull request merged**